src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetAutoMappedLocationsTable.tsx

```diff
@@ -1,61 +1,206 @@
-import ButtonText from '@common/components/ButtonText';
+import { PendingLocationMappingUpdate } from '@admin/pages/release/data/ReleaseApiDataSetLocationsMappingPage';
+import ApiDataSetLocationMappingModal from '@admin/pages/release/data/components/ApiDataSetLocationMappingModal';
+import getApiDataSetLocationCodes from '@admin/pages/release/data/utils/getApiDataSetLocationCodes';
 import Tag from '@common/components/Tag';
+import ButtonText from '@common/components/ButtonText';
+import Pagination from '@common/components/Pagination';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import FormSearchBar from '@common/components/form/FormSearchBar';
 import VisuallyHidden from '@common/components/VisuallyHidden';
-import React from 'react';
+import useDebouncedCallback from '@common/hooks/useDebouncedCallback';
+import { LocationLevelKey } from '@common/utils/locationLevelsMap';
+import chunk from 'lodash/chunk';
+import React, { useMemo, useState } from 'react';
+
+const locationsPerPage = 10;
+const formId = 'auto-mapped-search';
 
 interface Props {
+  level: LocationLevelKey;
   locations: AutoMappedLocation[];
+  newLocations: LocationCandidateWithKey[];
+  pendingUpdates?: PendingLocationMappingUpdate[];
+  onUpdate: (update: PendingLocationMappingUpdate) => Promise<void>;
 }
 
 export default function ApiDataSetAutoMappedLocationsTable({
+  level,
   locations,
+  newLocations,
+  pendingUpdates = [],
+  onUpdate,
 }: Props) {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [searchTerm, setSearchTerm] = useState<string>();
+
+  const filteredLocations = useMemo(() => {
+    return searchTerm
+      ? locations.filter(location => {
+          const { candidate, mapping } = location;
+          const searchFields = [
+            candidate.label.toLowerCase(),
+            candidate.code?.toLowerCase(),
+            candidate.laEstab?.toLowerCase(),
+            candidate.oldCode?.toLowerCase(),
+            candidate.ukprn?.toLowerCase(),
+            candidate.urn?.toLowerCase(),
+            mapping.source.label.toLowerCase(),
+            mapping.source.code?.toLowerCase(),
+            mapping.source.laEstab?.toLowerCase(),
+            mapping.source.oldCode?.toLowerCase(),
+            mapping.source.ukprn?.toLowerCase(),
+            mapping.source.urn?.toLowerCase(),
+          ];
+
+          return searchFields.some(field => {
+            return field?.includes(searchTerm.toLowerCase());
+          });
+        })
+      : locations;
+  }, [locations, searchTerm]);
+
+  const filteredLocationChunks = useMemo(
+    () => chunk(filteredLocations, locationsPerPage),
+    [filteredLocations],
+  );
+
+  const totalFilteredLocations = filteredLocations.length;
+  const totalPages = filteredLocationChunks.length;
+
+  const [handleSearch] = useDebouncedCallback((term: string) => {
+    setSearchTerm(term);
+    setCurrentPage(1);
+  }, 200);
```

Comment:
Would bump this up a fair whack (e.g. 800ms) as slow typists could end up searching more often than is necessary. 

For searching through schools, this could be lead to a really janky experience.